### PR TITLE
[WIP] iOS: Add copy to pasteboard action

### PR DESCRIPTION
**Pull request by DmitryPonomarenko**

### CODE REVIEW

File: detox/test/e2e/a-pasteboard.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+describe('Sanity', () => {
```
Comment (rotemmiz) [inline detox/test/e2e/a-pasteboard.js]: Change the describe text

File: detox/ios/Detox/DTXPasteboardInfo.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Dmytro Ponomarenko on 4/9/18.
+//  Copyright © 2018 Wix. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@interface DTXPasteboardInfo : NSObject
+
+- (NSDictionary *)pasteboardHaveSomeValue;
```
Comment (LeoNatan) [inline detox/ios/Detox/DTXPasteboardInfo.h]: Please give this method a more meaningful name like `pasteboardInfo`.

File: detox/ios/Detox/DTXPasteboardInfo.m [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+//  Created by Dmytro Ponomarenko on 4/9/18.
+//  Copyright © 2018 Wix. All rights reserved.
+//
+
+#import "DTXPasteboardInfo.h"
+#import <UIKit/UIKit.h>
+
+@implementation DTXPasteboardInfo
+
+- (NSDictionary *)pasteboardHaveSomeValue {
```
Comment (LeoNatan) [inline detox/ios/Detox/DTXPasteboardInfo.m]: Any reason this is an instance method rather than a class method?

File: detox/ios/Detox/DetoxManager.m [hunk 4]
```diff
@@ -78,6 +80,7 @@ - (instancetype)init
 	self.webSocket.delegate = self;
 	self.testRunner = [[TestRunner alloc] init];
 	self.testRunner.delegate = self;
+	self.pasteboardInfo = [[DTXPasteboardInfo alloc] init];
```
Comment (LeoNatan) [inline detox/ios/Detox/DetoxManager.m]: Let's make it a static method so we don't need to make an instance here.

File: detox/test/e2e/0-pasteboard.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+describe('Pasteboard', () => {
```
Comment (LeoNatan) [inline detox/test/e2e/0-pasteboard.js]: If there is t-pasteboard.js, why is this file needed?

File: detox/test/e2e/t-pasteboard.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  });
+
+  beforeEach(async () => {
+    await element(by.text('Pasteboard')).tap();
+  });
+
+  it('should have stringTextInput', async () => {
+    await element(by.id('stringValueInput')).typeText('exampleString\n');
+    await element(by.id('CheckButton')).tap();
+    await device.pasteboardInfo();
```
Comment (LeoNatan) [inline detox/test/e2e/t-pasteboard.js]: I don't understand. What does this check? This returns a JSON with the pasteboard info...—and?

File: detox/test/src/Screens/index.js [hunk 7]
```diff
@@ -13,6 +13,7 @@ import AnimationsScreen from './AnimationsScreen';
 import LocationScreen from './LocationScreen';
 import ShakeScreen from './ShakeScreen';
 import DatePickerScreen from './DatePickerScreen'
+import PasteBoardScreen from './PasteBoardScreen';
```
Comment (LeoNatan) [inline detox/test/src/Screens/index.js]: This should be `PasteboardScreen` not `PasteBoardScreen`

File: detox/test/e2e/t-pasteboard.js (hunk 6 above)
Comment (LeoNatan) [reply detox/test/e2e/t-pasteboard.js]: So basically I see missing API to `expect` on the pasteboard. `pasteboardInfo()` should be an internal function, not external.

File: detox/test/src/Screens/PasteBoardScreen.js [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+const boyImage = 'http://img.draugas.lt/forumas/veidukai/973676.gif'
+const redPandaImage = 'https://orig00.deviantart.net/ee08/f/2009/073/e/d/free_red_panda_icon_100x100_by_supertuffpinkpuff.png'
+const supermanImage = 'https://a.facdn.net/1484483600/perkalator445.gif'//'http://it.avatarstock.com/img/Superman-Logo_5194.jpg'
```
[the remark this answers is not in the document]
Comment (DmitryPonomarenko) [reply detox/test/src/Screens/PasteBoardScreen.js]: links, png will be deleted

File: detox/src/ios/expect.js [hunk 9]
```diff
@@ -369,6 +409,7 @@ class WaitForElement extends WaitFor {
 
 function expect(element) {
   if (element instanceof Element) return new ExpectElement(element);
+  if (element instanceof Device) return new ExpectDevice(element);
```
[the remark this answers is not in the document]
Comment (LeoNatan) [reply detox/src/ios/expect.js]:
```
await expect(device.pasteboard()).toHaveString(string);
await expect(device.pasteboard()).toHaveURL(url);
await expect(device.pasteboard()).toHaveImage();
await expect(device.pasteboard()).toHaveColor();
```

File: detox/test/e2e/t-pasteboard.js [hunk 10]
```diff
@@ -10,29 +10,29 @@ describe('Pasteboard', () => {
...
     await element(by.id('testImageValue')).tap()
     await element(by.id('CheckButton')).tap()
-    await expect(device).pasteboardToHaveImage();
+    await expect(await device.pasteboard()).toHaveImage();
     await element(by.id('backButton')).tap()
   });
 
   it('should have urlValue', async () => {
     await element(by.id('testURLValue')).tap();
     await element(by.id('testURLValue')).typeText('exampleURL\n');
```
Comment (rotemmiz) [inline detox/test/e2e/t-pasteboard.js]: is that a URL or string? this test seems weird. This is not a URL, yet pasteboard returns true for `toHaveURL`

**Comment by rotemmiz** [comment]
API makes much more sense now, good progress

File: detox/src/ios/expect.js [hunk 11]
```diff
@@ -303,8 +305,41 @@ class Element {
   }
 }
 
+
 class Expect { }
 
+class PasteboardInfo {
```
Comment (rotemmiz) [inline detox/src/ios/expect.js]: This should be in a separate class file
